pkg/webhook/elasticquota/quota_topology_check.go

```diff
@@ -63,12 +63,10 @@ func (qt *quotaTopology) validateQuotaSelfItem(quota *v1alpha1.ElasticQuota) err
 }
 
 // validateQuotaTopology checks the quotaInfo's topology with its parent and its children (when update calls the function).
-func (qt *quotaTopology) validateQuotaTopology(oldQuotaInfo, quotaInfo *QuotaInfo) error {
-	// interchange between parentQuotaGroup and childQuotaGroup is prohibited now.
-	if oldQuotaInfo != nil && oldQuotaInfo.IsParent != quotaInfo.IsParent {
-		return fmt.Errorf("IsParent is forbidden modify now, quotaName:%v", oldQuotaInfo.Name)
+func (qt *quotaTopology) validateQuotaTopology(oldQuotaInfo, quotaInfo *QuotaInfo, oldQuota *v1alpha1.ElasticQuota) error {
+	if err := qt.checkIsParentChange(oldQuotaInfo, quotaInfo, oldQuota); err != nil {
```

Comment:
Maybe we should move the check after line 73 and still refuse to modify the root quota.

Reply:
root quota is forbidden modify in function IsForbiddenModify. If the check is after line 73, it will skip the check when quota IsParent change from "false" to "true". But as discussed, IsParent change from "false" to "true" need the quota has no bound pods.